homeassistant/components/screenlogic/entity.py

```diff
@@ -109,7 +109,7 @@ def _async_data_updated(self) -> None:
 
     async def async_added_to_hass(self) -> None:
         """When entity is added to hass."""
-
+        await super().async_added_to_hass()
         self.async_on_remove(
             await self.gateway.async_subscribe_client(
                 self._async_data_updated, self._update_message_code
```

Comment:
```suggestion
                self._async_data_updated_push, self._update_message_code
```

And than you can make `_async_data_updated` only write if available changes

Reply:
Not sure if we need to consider out of order callbacks as well

Get push updated, write state
Coordinator poll update finishes, writes state

Can the coordinator update callback ever be older than the push update?

Reply:
> Rename the existing `_async_data_updated` to `_async_data_updated_push` to preserve the existing functionality
> 
> And than you can make `_async_data_updated` only write if a new `self._last_update_success != self.coordinator.last_update_success` changes and change `self._last_update_success` every time we write there

`_async_data_updated` is only used by the push functionality.  `_handle_coordinator_update` is the callback for the coordinator.  Could I just override `_handle_coordinator_update` locally with the last update check?

> Not sure if we need to consider out of order callbacks as well
> 
> Get push updated, write state Coordinator poll update finishes, writes state
> 
> Can the coordinator poll update callback ever be older than the push update?

It shouldn't matter the order at which callbacks are called. All entities get their state directly from the api data, which is updated as each response is received. Entities that are pushed don't look at polled data, and vise-versa. `available` always points to `self.coordinator.last_update_success`. 
A push update could trigger `_async_data_updated` before the coordinator calls `_handle_coordinator_update`, but either way they are getting the latest from `gateway.get_data()` and `self.coordinator.last_update_success`. I think the most we may need to do is update `self._last_update_success` in both callbacks.

Reply:
I think [bb36b4f](https://github.com/home-assistant/core/pull/89832/commits/bb36b4faf77ab9380b367bf2e2c719c4d30eeb2f) resolves this.